src/ui/carousel/carousel.ts

```diff
@@ -0,0 +1,614 @@
+   * @param slide
+   */
+  getSlidesAfter(slide: HTMLElement): HTMLElement[] {
+    if (this.loop) {
+      return this.splitSlidesInHalf(slide, Direction.RIGHT);
+    } else {
+      return this.getSlides().slice(this.getSlides().indexOf(slide) + 1);
+    }
+  }
+
+  /**
+   * Go to the next slide.
+   */
+  next(): void {
+    this.transitionSlidesBy(1);
+  }
+
+  /**
+   * Go to the previous slide.
+   */
+  prev(): void {
+    this.transitionSlidesBy(-1);
+  }
+
+  /**
+   * Transition to the slide with the given index.
+   */
+  goToIndex(index: number, drivenBySync = false): void {
+    const clampedIndex = this.getClampedIndex(index);
+    this.goToSlide(this.getSlideByIndex(clampedIndex), drivenBySync);
+  }
+
+  /**
+   * Register a function to be called when the given event is fired.
+   */
+  on(event: CarouselEvent, callback: (data: CarouselEventData) => void) {
+    this.eventManager.on(event, callback);
+  }
+
+  /**
+   * Remove a function that is called when the given event is fired.
+   */
+  off(event: CarouselEvent, callback: (data: CarouselEventData) => void) {
+    this.eventManager.off(event, callback);
+  }
+
+  /**
+   * Return the slide element at the given index in the list of slides.
+   * @param index
+   */
+  getSlideByIndex(index: number): HTMLElement {
+    return this.slides[index];
+  }
+
+  /**
+   * Return the number of slides.
+   */
+  getSlideCount(): number {
+    return this.slides.length;
+  }
+
+  /**
+   * Dispose of the carousel.
+   */
+  dispose() {
+    this.raf.dispose();
+    this.transition.dispose();
+  }
+
+  /**
+   * Clear the transition target, stop transitioning.
+   */
+  stopTransition(): void {
+    this.transitionTarget = null;
+  }
+
+  /**
+   * Clamp a given index to land within the possible slide indices.
+   * @param index
+   */
+  getClampedIndex(index: number): number {
+    const slidesLength = this.getSlides().length;
+    if (this.loop) {
+      return mathf.wrap(index, 0, slidesLength);
+    } else {
+      return mathf.clamp(0, slidesLength - 1, index);
+    }
+  }
+
+  /**
+   * Synchronize this carousel with the given carousels.
+   */
+  sync(...carousels: Carousel[]): void {
+    const masterSet = setf.merge(
+        this.syncedCarousels,
+        ...carousels.map((c) => c.syncedCarousels));
+    this.syncedCarousels = masterSet;
+    masterSet.forEach((c) => c.syncedCarousels = masterSet);
+  }
+
+  /**
+   * Transition to the slide `value` slides away.
+   * @param value
+   */
+  private transitionSlidesBy(value: number): void {
+    const nextIndex =
+        this.getSlides()
+            .indexOf(this.transitionTarget || this.getActiveSlide()) +
+        value;
+    this.goToIndex(nextIndex);
+  }
+
+  /**
+   * Sync this carousel to the given index in the given carousel.
+   * @param rawIndex
+   * @param carousel
+   * @private
+   */
+  private syncTo(rawIndex: number, carousel: Carousel) {
+    /**
```

Comment:
FYI The `/**` format for comments should be reserved for docstrings. You might be able to move this comment block into the main docstring since it's so long.

Reply:
Done.